datasets/poleval2019_mt/poleval2019_mt.py

```diff
@@ -0,0 +1,206 @@
+                "target_file": path_tmpl.format(dl_dir=path_to_manual_file, split=split, lang=target),
+                "split": split,
+            }
+
+        # To handle test split when english is the target language.
+        # This is because there is no Polish to English test file that is available in the default set
+        if "en" == source:
+            dl_dir_test = dl_manager.download_and_extract(_TEST_URL)
+            test_file = "{dl_dir}/task4_test/tst.en".format(dl_dir=dl_dir_test)
+        elif "en" == target:
+            test_file = ""
+        else:
+            dl_dir_test = dl_manager.download_and_extract(_TEST_URL)
+            test_file = test_tmpl.format(dl_dir=dl_dir_test, target=target.upper(), source=source)
+
+        files["test"] = {"source_file": test_file, "target_file": "", "split": "test"}
+
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs=files["train"]),
+            datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs=files["dev"]),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs=files["test"]),
+        ]
+
+    def _generate_examples(self, source_file, target_file, split):
+        """This function returns the examples in the raw (text) form."""
+        source, target = self.config.language_pair
+
+        # Returning an empty source and target just to handle the test file absence when English is the target
+        if split == "test":
+            if target == "en":
+                # Returning dummy info
+                result = {"translation": {source: "", target: ""}}
+                yield 0, result
+            else:  # Handling cases for Polish and Russian languages
+                with open(source_file, encoding="utf-8") as f:
+                    source_sentences = f.read().split("\n")
+
+                for idx, sent in enumerate(source_sentences):
+                    if sent.strip() != "":
+                        result = {"translation": {source: sent, target: ""}}
+                        yield idx, result
+        else:
+            # Training and Dev sets examples
+            with open(source_file, encoding="utf-8") as f:
+                source_sentences = f.read().split("\n")
+            with open(target_file, encoding="utf-8") as f:
+                target_sentences = f.read().split("\n")
+
+            assert len(target_sentences) == len(source_sentences), "Sizes do not match: %d vs %d for %s vs %s." % (
+                len(source_sentences),
+                len(target_sentences),
+                source_file,
+                target_file,
+            )
+
+            for idx, (l1, l2) in enumerate(zip(source_sentences, target_sentences)):
+                result = {"translation": {source: l1, target: l2}}
+                # Make sure that both translations are non-empty.
+                if all(result.values()):
+                    yield idx, result
```

Comment:
```suggestion
                yield idx, result
```
You can remove the if statement.

Indeed since `result ` is never an empty dict (it has a "translation key), then `all(result.values())` is always true

Reply:
That's dumb of me! Will work on all changes and push the code again. Thanks!